Automattic/jetpack
# Jetpack plugin: Require and use jetpack-masterbar package

Load the Masterbar module (WordPress.com Toolbar) functionality from the `masterbar-package` instead of using the corresponding Module code, soon to be deprecated.

## Proposed changes:

* `jetpack\modules\masterbar.php`: Use package to load the Masterbar module functionality 
* Backport modules updates from https://github.com/Automattic/jetpack/pull/37625

### Other information:


## Jetpack product discussion


pfwV0U-3U-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






**Important!** While testing on WoA make sure to also activate the WordPress.com Features plugin (Bleeding Edge just in case) via Beta Tester and add `define( 'JETPACK_MU_WPCOM_LOAD_VIA_BETA_PLUGIN', true );` to your `wp-config.php`

### nav-unification: Testing the admin menu in Simple, WoA and self-hosted:
You can test this in the [WPCOM REST API Dev console](https://developer.wordpress.com/docs/api/console/) for the `wpcom/v2/sites/YOUR_BLOG_ID/admin-menu` endpoint. 

**For block based themes, "Customize" should not be present under "Appearance".**

Make sure to test with/without the PR applied and confirm the responses are identical. For Simple and WoA sites we'll also need to test with classic/default Admin Interface Style.

### nav-unification: Admin color schemes

For self-hosted sites and **classic** WoA sites, upon editing a User Profile via wp-admin, confirm the Admin Color Scheme field is present:

<img width="752" alt="Screenshot 2024-06-05 at 10 20 28" src="https://github.com/Automattic/jetpack/assets/1758399/3d8846b3-ee5b-469f-9b9a-7bcf2fbab927">

Ensure that by picking one of the options, the admin theme is updated. 

For **default** WoA sites, upon editing a User Profile via wp-admin, confirm you see the following link to update the field on WPCOM:
<img width="534" alt="Screenshot 2024-06-06 at 10 16 48" src="https://github.com/Automattic/jetpack/assets/1758399/ccedb1b7-76ae-446d-8272-624d7a1f804c">

**For a WoA site, this change should be also reflected in Calypso.**

Confirm that the `admin_color` user meta is also part of the corresponding REST API  response to the `users` endpoint. 
You can test this in the [WPCOM REST API Dev console](https://developer.wordpress.com/docs/api/console/) for the `wp/v2/sites/YOUR_BLOG_ID/users` endpoint. 

Make sure to test this with a self-hosted and WoA. Not applicable for Simple sites.

### Classic vs Default view

Try switching between Classic vs Default view for a WoA and Simple site. Confirm nothing breaks and both views are displayed as expected.
<img width="334" alt="Screenshot 2024-06-04 at 17 01 30" src="https://github.com/Automattic/jetpack/assets/1758399/cb64d171-e04d-47e2-b6d5-e722b16e7c16">

For self-hosted sites, this option should NOT be visible since there should be [no nav customizations on WP Admin of Jetpack sites](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/masterbar/src/admin-menu/load.php#L34)

For WoA sites with SSO disabled this option should NOT be visible since there should be [no nav customizations on WP Admin of Atomic sites when SSO is disabled.](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/masterbar/src/admin-menu/load.php#L29C8-L29C76)

### Editing a user profile on wp-admin for WoA sites

When editing a user's profile via wp-admin on a WoA site, confirm that you see the following message and the corresponding links to your WPCOM account page work as expected:

<img width="1020" alt="Screenshot 2024-06-05 at 10 02 12" src="https://github.com/Automattic/jetpack/assets/1758399/75201f5d-51f1-4f3b-8687-85849d2c551e">

### Inline Help
TBA

### Nudges to purchase a WordPress.com Plan to be able to customize CSS on your site (Simple and WoA sites)
Make sure your site is 1) using a **non**-block theme, eg Hever. and 2) IS NOT on the explorer plan
Navigate to Appearance -> Additional CSS and confirm you see the following nudge:

<img width="441" alt="Screenshot 2024-06-05 at 11 08 09" src="https://github.com/Automattic/jetpack/assets/1758399/d1f52cfc-66ae-4983-b72c-e725a924918f">

Ensure the Upgrade flow is working as expected by clicking the "Upgrade now" button. You should be redirected to checkout with the Explorer plan in your cart.
_Note sure how to test this on a WoA site_

### Disable edit_post and delete_post capabilities for Posts Pages in WP-Admin and display a notification icon.
See https://github.com/Automattic/jetpack/pull/19900

### RTL support
**Simple & WoA**: Update your WPCOM user locale to Arab and ensure the Masterbar and Admin menu is displayed consistently with/without the PR.
**Self hosted**: Update your user locale to Arab and ensure the Masterbar and Admin menu is displayed consistently with/without the PR.
Let's make sure to test this as much as possible since the package loads RTL styles automatically via the Assets package.
Example: All the relevant `$this->is_rtl()` conditionals in the [module code](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/modules/masterbar/admin-menu/class-base-admin-menu.php#L254) have been removed from the corresponding package code.
